components/ILIAS/Saml/classes/class.ilSamlSettingsGUI.php

```diff
@@ -494,26 +495,19 @@ private function showUserAttributeMappingForm(?ilPropertyFormGUI $form = null):
         $this->tpl->setContent($form->getHTML());
     }
 
-    private function getSettingsForm(): ilPropertyFormGUI
+    private function getSettingsForm(array $values = []): StandardForm
     {
-        $form = new ilPropertyFormGUI();
-        $form->setFormAction($this->ctrl->getFormAction($this, self::CMD_SAVE_SETTINGS));
-        $form->setTitle($this->lng->txt('auth_saml_configure'));
-
-        $show_login_form = new ilCheckboxInputGUI($this->lng->txt('auth_saml_login_form'), self::LNG_LOGIN_FORM);
-        $show_login_form->setInfo($this->lng->txt('auth_saml_login_form_info'));
-        $show_login_form->setValue('1');
-        $form->addItem($show_login_form);
-
-        if (!$this->access->checkAccess(self::PERMISSION_WRITE, '', $this->ref_id)) {
-            foreach ($form->getItems() as $item) {
-                $item->setDisabled(true);
-            }
-        } else {
-            $form->addCommandButton(self::CMD_SAVE_SETTINGS, $this->lng->txt(self::CMD_SAVE));
-        }
-
-        return $form;
+        return $this->ui_factory->input()->container()->form()->standard(
+            $this->ctrl->getFormAction($this, self::CMD_SAVE_SETTINGS),
+            [
+                self::LNG_LOGIN_FORM => $this->ui_factory->input()->field()->checkbox(
+                    $this->lng->txt('auth_saml_login_form'),
+                    $this->lng->txt('auth_saml_login_form_info')
+                )
+                    ->withValue((bool) ($values[self::LNG_LOGIN_FORM] ?? true))
+                    ->withDisabled(!$this->access->checkAccess(self::PERMISSION_WRITE, '', $this->ref_id)),
```

Comment:
It was wrong before, but we need to use `rbac_system` here to check the access. The form is located in the "Administratiuon" area, so we must not use `ilAccessHandler` here (which perform "Tree Checks").
Please also fix the other usages of `$this->access->` and remove this class property afterwards.

Furthermore. Please rename the "Submit Button Label" to `$this->lng->txt('refresh')` in case the user has no `write` permission.